src/ui/public/index_patterns/index_patterns.js

```diff
@@ -41,6 +41,7 @@ export function IndexPatternsProvider(Notifier, Private) {
   self.cache = patternCache;
   self.getIds = getProvider('id');
   self.getTitles = getProvider('attributes.title');
+  self.getIndexPatterns = getProvider();
```

Comment:
I think it's still weird because this new method is the base method and the others are built on top of its response so I feel the names should reflect that in someway.

You'll need a second reviewer for this PR. Once you add them, let's see what they think